main.cpp

```diff
@@ -64,64 +64,120 @@ send me a DM to check your pull request
-    //2
-    //3
+    //Initialize "value" and "name" 
+    float value;
+    std::string name;
+
+    T(int v, const char* myName) :  //1 //Constructs that assigns Constructor Parameters to variables above
+    value (v), //2 
+    name (myName)//3
+    {}
 };
 
-struct <#structName1#>                                //4
+struct CompareFunc                                //4
 {
-    <#type name#> compare(<#type name#> a, <#type name#> b) //5
+    T* compare(T* a, T* b) //5
     {
-        if( a->value < b->value ) return a;
-        if( a->value > b->value ) return b;
-        return nullptr;
+        if (a != nullptr && b != nullptr)
+        {
+            //Compare the Value (float type) from the two Pointers and return the smaller one
+            if( a->value < b->value ) return a; //compare the two value contained in the    pointers to Type "T" created above
+            if( a->value > b->value ) return b;
+        }
+        return nullptr; //if none of the if statements matched return a nullptr
     }
 };
 
 struct U
 {
-    float <#name1#> { 0 }, <#name2#> { 0 };
-    <#returnType#> <#memberFunction#>(<#type name#>* <#updatedValue#>)      //12
+    U(){std::cout << "New Object of Type U has been created" << std::endl;}
+    
+    float uSetPoint { 0 }, uProcessVariable { 0 };
+
+    float updateLogic(float* newValue)      //12
     {
-        
+        if (newValue == nullptr)
+        {
+            std::cout<<"Detected Nullpointer!" << std::endl;
+            return 0.0f; 
+        }
+
+        std::cout << "U's uSetPoint value: " << this->uSetPoint << std::endl;
+        this->uSetPoint = *newValue; //Acess the uSetPoint float that is part of the Pointer to a Type U* named that
+        std::cout << "U's uSetPoint updated value: " << this->uSetPoint << std::endl;
+
+        while( std::abs(this->uSetPoint - this->uProcessVariable) > 0.001f)
+        {
+            this->uProcessVariable += 0.5f ;
+            std::cout << this->uProcessVariable << std::endl;
+        }
+        std::cout << "U's uProcessVariable updated value: " << this->uProcessVariable << std::endl;
+        return this->uProcessVariable * this->uSetPoint;
     }
 };
 
-struct <#structname2#>
+struct MyUpdater
 {
-    static <#returntype#> <#staticFunctionA#>(U* that, <#type name#>* <#updatedValue#> )        //10
+    static float updateLogic(U* that, float* newValue )        //10
     {
-        std::cout << "U's <#name1#> value: " << that-><#name1#> << std::endl;
-        that-><#name1#> = <#updatedValue#>;
-        std::cout << "U's <#name1#> updated value: " << that-><#name1#> << std::endl;
-        while( std::abs(that-><#name2#> - that-><#name1#>) > 0.001f )
+        if (newValue == nullptr && newValue == nullptr)
         {
-            /*
-             write something that makes the distance between that-><#name2#> and that-><#name1#> get smaller
-             */
-            that-><#name2#> += ;
+            std::cout<<"Detected Nullpointer!" << std::endl;
```

Comment:
add space around your `<<` to improve readability
- [ ] revise

also, you should return 0 if both function params are nullptr.